data/completion/etelpmoc.sh

```diff
@@ -202,10 +202,19 @@ esac
 
 if [ ! "$_bounce" ]; then
     if [ "$_compact" ]; then
-        readarray -t COMPREPLY < <( builtin compgen -A "$_compact" -- "${COMP_WORDS[$COMP_CWORD]}" )
+        readarray -t COMPREPLY < <( builtin compgen -A "$_compact" -- "${COMP_WORDS[COMP_CWORD]}" )
     elif [ "$_compfunc" ]; then
         # execute completion function (or the command if -C)
-        $_compfunc
+
+        # from https://www.gnu.org/software/bash/manual/html_node/Programmable-Completion.html:
+        #   When the function or command is invoked, the first argument ($1) is
+        #   the name of the command whose arguments are being completed, the
+        #   second argument ($2) is the word being completed, and the third
+        #   argument ($3) is the word preceding the word being completed on the
+        #   current command line.
+        # that's "$1" "${COMP_WORDS[COMP_CWORD]}" and "${COMP_WORDS[COMP_CWORD-1]}"
+        # (probably)
+        $_compfunc "$1" "${COMP_WORDS[COMP_CWORD]}" "${COMP_WORDS[COMP_CWORD-1]}"
```

Comment:
Do you know whether COMP_CWORD is always > 0 when this is called?

Reply:
yes, because `COMP_WORDS[0]` is the command you're completing (so you can't get here without `COMP_CWORD` being at least 1)